apps/web/modules/event-types/views/event-types-single-view.tsx

```diff
@@ -196,6 +203,31 @@ const EventTypePage = (props: EventTypeSetupProps) => {
     },
   });
 
+  const router = useRouter();
+
+  useEffect(() => {
+    const handleRouteChange = (url: string) => {
+      const paths = url.split("/");
+      if (
+        eventType.hosts.length === 0 &&
+        !leaveWithoutAssigningHosts.current &&
+        (url === "/event-types" || paths[1] !== "event-types")
```

Comment:
If we want to show warning dialog only when navigating to a non `/event-types` route, do we need the `url === "/event-types"` check on line 214?

Reply:
This only checks the back button, because the route is also /event-types.